### ISSUE TITLE
Flatpak Support

### ISSUE THREAD

**Issue by Anvil5465** [mullvad/mullvad-browser#6]
It would be really nice to have this packaged as a flatpak, maybe on [Flathub](https://flathub.org), for easier installation and software updates across linux distros.

**Comment by Tc-001** [comment]
Also you can register (and relatively easily add) custom flatpak repositories, so it could still be served from Mullvad's servers.

**Comment by Anderhar** [comment]
I was going to open the exactly same issue, but you beat me to it by half an hour! 
I've been using Tor Browser disconnected from Onion network and also with the uBlock extension for several years now, but I install it as Flatpak version of [Tor Browser Launcher](https://flathub.org/apps/details/com.github.micahflee.torbrowser-launcher), and this is a major difference from the Mullvad's edition, which announcement greatly encouraged me.

**Comment by ruihildt** [comment]
Thanks for your feedback.
We will look into further distributing Mullvad Browser at some point in the future.

**Comment by nixfu** [comment]
Yep, Flatpak is the main method now to distribute Linux software, that is flavor independent.  It is by far the best way to distribute this browser.

**Comment by storopoli** [comment]
Yes, flatpak!

**Comment by Thorin-Oakenpants** [comment]
make sure RLBox is implemented - https://gitlab.com/librewolf-community/browser/source/-/issues/9#note_1345338224

I believe @PieroV has been working in this area lately :)

**Comment by jcnafi** [comment]
Yeah, we need a flatpak version. I use all my browsers as flatpak.

**Comment by dalaeance** [comment]
Throwing up a +1 for Flatpak support. If I gotta do it from source, it pretty much never gets used and/or updated after the first few uses.

**Comment by polarhive** [comment]
I like keeping my $HOME folder clean: Firefox based browsers don't respect the [xdg-basedir](https://wiki.archlinux.org/title/XDG_Base_Directory#Supported) spec by default.

It creates a .mozilla folder to store browser profiles. As a workaround: I use Flatseal to limit filesystem access for $HOME. The flatpak variants of Firefox, LibreWolf & TBB then store the profile files under ~/.var/app/{flatpak app-id}. 

+1 for flatpak support

**Comment by storopoli** [comment]
The flatpak app was merged and should be available soon.
https://github.com/[flathub/net.mullvad.MullvadBrowser](https://github.com/flathub/net.mullvad.MullvadBrowser)

**Comment by jcnafi** [comment]
> The flatpak app was merged and should be available soon. https://github.com/flathub/net.mullvad.MullvadBrowser

That's a good thing to hear. Never thought it would be so quick.

**Comment by birds-swim** [comment]
Thank you!  Hats off to the Hero Devs of Mullvad!

**Comment by ruihildt** [comment]
FYI this is not an official release and we haven't verified it.
We are currently considering how to best handle this.

**Comment by tinypinkdragons** [comment]
As soon as I open the flatpak, my Pi-hole registers a connection attempt to 'aax-us-pdx.amazon-adsystem.com', which the Windows version doesn't do. After some investigation, I came here and saw it wasn't official.
Now uninstalled.
Any news on an official version, or will you not be publishing to Flathub?

**Comment by jcnafi** [comment]
isn't `net.mullvad.MullvadBrowser` official from flathub?

**Comment by tinypinkdragons** [comment]
> isn't `net.mullvad.MullvadBrowser` official from flathub?

As @ruihildt said, 

> FYI this is not an official release and we haven't verified it.

Unless there's been an update, and it hasn't been mentioned here, then I'm assuming it's still unofficial. I'd love to be corrected. Not sure why it's contacting an Amazon ad server if it is official.

**Comment by adrelanos** [comment]
@storopoli 
> The flatpak app was merged and should be available soon. https://github.com/flathub/net.mullvad.MullvadBrowser

This comment made it look as if it's official while it wasn't.

This was further exaggerated by:

> @Anvil5465 closed this as completed

Quote @ruihildt https://github.com/mullvad/mullvad-browser/issues/6#issuecomment-1536629529:
> FYI this is not an official release and we haven't verified it. We are currently considering how to best handle this.

Other stuff which was poorly handled in this ticket:

* @ruihildt not re-opening the ticket after commenting that it's unofficial.
* Mullvad not more quickly pointing out that it's unofficial.
* Mullvad neglecting to request flathub to remove or at least rename the unofficial Mullvad Browser. You should protect your trademark, reputation better.
  * Not even after @tinypinkdragons's comment that there might be adware injected (unexpected connections to amazon). If there might be adware, obviously there's also a risk for malware.

I've also created a ticket at flathub about this:

* https://github.com/flathub/net.mullvad.MullvadBrowser/issues/16

**Comment by ruihildt** [comment]
@adrelanos I could definitely have handled things better in regards to updating the status of the ticket and warning users that it was not an official package.

Regarding the Flathub page itself, it is indeed questionable that they replicated the Twitter like "blue tick for verified".
- See unofficial Google Chrome: https://flathub.org/apps/com.google.Chrome
- official Firefox: https://flathub.org/apps/org.mozilla.firefox
However, this is a Flathub general issue, not specific to this package.

We have plans to take over the package and make it official in the near future.

In the meantime, I'm submitting a PR to add metadata and information in the Flathub page signifying it's not an official package.

**Comment by adrelanos** [comment]
Your constructive response and prompt action is much appreciated!

**Comment by mlario** [comment]
Any news on flatpak side? would be much appreciated.

and would solve an issue of mullvad browser not being able to be recognized as an option for default browser under linux

**Comment by Ser-Bul** [comment]
FYI:
Recently Linux Mint stopped showing Unverified flatpaks in its Software Manager search results, making those unseen by default (https://www.phoronix.com/news/Linux-Mint-Unverified-Flatpaks)

**Comment by testuser51** [comment]
Are there any plans to officially support Flatpak? I don't want to install unverified stuff and I'm sure there are many people like me

**Comment by gitty323** [comment]
Hope to see official support on this soon, hosted on Mullvad flatpak servers like how you can host your own Fdroid repo on Android would be great. Additionally, I'd love to see a Mullvad Browser Alpha flatpak as I want to try the latest alpha that upgrades to Firefox 140 on the backend on my Bazzite install but do not see a straightforward way of doing that as I have to rely on the unofficial flatpak on an immutable distro.

**Comment by GuyDebord** [comment]
This is truly shameful for a privacy/security focused project after all these years to let an unverified third party distribute its app on flathub only because they are not willing to do it themselves.

A huge part of your Linux user base is unknowingly using a highly insecure Mullvad Browser and yet you still haven't done anything about it. This stance is incomprehensible.

Flatpak is the present and future of Linux applications. It offers developpers the opportunity to focus on a single format to be able to reach the entire Linux user base.
If ressources are a problem then simply drop the support for a restricting format in order to support a global one. Anyway your users' security should be a top priority.

I hope you will realise how not addressing this critical security issue discredit your stance on privacy/security and that you will finally take action.

**Comment by ruihildt** [comment]
@GuyDebord We have flatpak on our roadmap but there are other priorities at the moment, as you might have guessed.
The nature of an open source project means anyone can take the code and do what they want with it.
Similarly, there are also other community based packages for arch and for nix.

If users are not able to use the `.deb` or `.rpm` packages, there is the standalone installation which is working for all Linux distributions and is officially supported.

You can also track progress for this issue here: https://gitlab.torproject.org/tpo/applications/mullvad-browser/-/issues/207

**Comment by jonathancross** [comment]
1 year later... Link has changed: https://gitlab.torproject.org/tpo/applications/mullvad-browser/-/work_items/207
Status set to "Not Scheduled".
